kratos/mpi/tests/sources/test_mpi_communicator.cpp

```diff
@@ -178,6 +178,38 @@ KRATOS_TEST_CASE_IN_SUITE(MPICommunicatorSynchronizeAnd, KratosMPICoreFastSuite)
     KRATOS_CHECK_EQUAL(r_center.Is(PERIODIC), rank_is_even); // This one should be left untouched
 }
 
+KRATOS_TEST_CASE_IN_SUITE(MPICommunicatorSynchronizeNodalFlags, KratosMPICoreFastSuite)
+{
+    Model model;
+    ModelPart& r_model_part = model.CreateModelPart("TestModelPart");
+    r_model_part.AddNodalSolutionStepVariable(PARTITION_INDEX);
+
+    MPIDataCommunicator world_comm(MPI_COMM_WORLD);
+    Internals::ModelPartForMPICommunicatorTests(r_model_part, world_comm);
+
+    const int rank = world_comm.Rank();
+    const bool rank_is_even( (rank % 2) == 0 );
+
+    for (auto i_node = r_model_part.NodesBegin(); i_node != r_model_part.NodesEnd(); ++i_node)
```

Comment:
@jcotela why not adding it only on local nodes?

Reply:
It probably makes no difference, but in the test ModelPart one of the ghost nodes is always in rank i+1 (the other is the center node, on rank 0). Setting it to the local value (true if i is even) makes sure that the ghost node is set at the wrong value, and that the test must fail if communication goes wrong.

Reply:
ok :+1: